### PR TITLE
Feature/v1

### PR DESCRIPTION
**Pull request by rdenarie**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    <dependency>
+      <groupId>com.googlecode.java-diff-utils</groupId>
+      <artifactId>diffutils</artifactId>
+      <version>1.2</version>
+      <type>jar</type>
+      <scope>compile</scope>
+    </dependency>
+
+  </dependencies>
+</project>
```
Comment (thomasdelhomenie) [inline pom.xml:105]: Should we add the eXo maven repository here to make it easier for developers (no additional steps to make it work) ?

File: src/main/java/org/exoplatform/maven/plugin/DiffExtensionMojo.java [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+    /**
+     * Files to check
+     */
+    @Parameter(defaultValue="")
+    private List<FileToCheck> files;
+
+
+
+    public void execute() throws MojoExecutionException,MojoFailureException {
```
Comment (thomasdelhomenie) [inline src/main/java/org/exoplatform/maven/plugin/DiffExtensionMojo.java:109]: It would be good to add unit tests.

File: pom.xml (hunk 1 above)
Comment (rdenarie) [reply pom.xml:105]: which repository do you mean ?

Comment (thomasdelhomenie) [reply pom.xml:105]: The exo repository, like we did on addons, for example for task: https://github.com/exo-addons/task/blob/develop/pom.xml#L146-L173

File: src/main/java/org/exoplatform/maven/plugin/DiffExtensionMojo.java (hunk 2 above)
Comment (rdenarie) [reply src/main/java/org/exoplatform/maven/plugin/DiffExtensionMojo.java:109]: done

File: pom.xml (hunk 1 above)
Comment (rdenarie) [reply pom.xml:105]: what is the aim of this ?
As I understand, it is allow someone to build the project, without having repository declared in his maven settings, rigth ?
So, this should be at upper level, no ? in this artifact for example :
<artifactId>addons-parent-pom</artifactId>
    <groupId>org.exoplatform.addons</groupId>
    <version>12-M01</version>

Comment (thomasdelhomenie) [reply pom.xml:105]: > As I understand, it is allow someone to build the project, without having repository declared in his maven settings, rigth ?

Right.

> So, this should be at upper level, no ?

Yes we can but it requires to inherit from the addon parent pom, which is not the case here.
But not a big deal, it is such a suggestion to improve dev experience of potential external contributors.

Comment (rdenarie) [reply pom.xml:105]: done

**Review by thomasdelhomenie — APPROVED** [review]

**Pull request merged**